### PR TITLE
Core/PacketIO: Enabled and updated Mail [WIP]

### PR DESCRIPTION
**Pull request by opcod3**

### CODE REVIEW

File: src/server/game/Handlers/MailHandler.cpp [hunk 1]
```diff
@@ -582,8 +582,10 @@ void WorldSession::HandleGetMailList(WorldPackets::Mail::MailGetList& packet)
     for (Mail* m : player->GetMails())
     {
         // skip deleted or not delivered (deliver delay not expired) mails
-        if (m->state == MAIL_STATE_DELETED || cur_time < m->deliver_time)
+        if (m->state == MAIL_STATE_DELETED || cur_time < m->deliver_time){
+            response.TotalNumRecords--;
```
Comment (Gigi1237) [inline src/server/game/Handlers/MailHandler.cpp]: Is this a hack? It feels like one.

Comment (joschiwald) [inline src/server/game/Handlers/MailHandler.cpp]: this one should be better
```c++
WorldPackets::Mail::MailListResult response;
time_t curTime = time(nullptr);

for (Mail* m : player->GetMails())
{
    // skip deleted or not delivered (deliver delay not expired) mails
    if (m->state == MAIL_STATE_DELETED || curTime < m->deliver_time)
        continue;

    // max. 50 mails can be sent
    if (response.Mails.size() < 50)
        response.Mails.emplace_back(m, player);

    ++response.TotalNumRecords;
}
```

File: src/server/game/Handlers/MailHandler.cpp [hunk 2]
```diff
@@ -585,11 +584,11 @@ void WorldSession::HandleGetMailList(WorldPackets::Mail::MailGetList& packet)
         if (m->state == MAIL_STATE_DELETED || cur_time < m->deliver_time)
             continue;
 
-        response.Mails.emplace_back(m, player);
-
         // max. 50 mails can be sent
-        if (response.Mails.size() >= 50)
-            break;
+        if (response.Mails.size() < 50)
```
Comment (jackpoz) [inline src/server/game/Handlers/MailHandler.cpp]: are the changes in this file required by this PR or are they just refactoring/codestyle changes ?

Comment (Gigi1237) [inline src/server/game/Handlers/MailHandler.cpp]: They are required changes.

Before TotalNumRecords included pending mails, this makes sure it does not. Making it include pending mails causes the client to complain about the mailbox being full.

Comment (jackpoz) [inline src/server/game/Handlers/MailHandler.cpp]: so that's actually a fix and should be applied to branch 335 too ?

Comment (Gigi1237) [inline src/server/game/Handlers/MailHandler.cpp]: I don't know if it applies to 3.3.5. Never worked on it.

Comment (joschiwald) [inline src/server/game/Handlers/MailHandler.cpp]: not required on 335, in that code it was handled correctly

**Comment by Carbenium** [comment]
Could you please remove the opcode value changes since they aren't really up to date anymore ;)

**Pull request closed without merging**